### ISSUE TITLE
Transcription breaks error stream when hosted in .Net Core on IIS

### ISSUE THREAD

**Issue by pcarnella** [PowerShell/PowerShell#14734]
## Steps to reproduce

Create ASP.Net Core app targeting .Net 5.0 (I saw same problem with 3.1).  Add the PowerShell packages:

    <PackageReference Include="Microsoft.Management.Infrastructure" Version="2.0.0" />
    <PackageReference Include="Microsoft.Management.Infrastructure.CimCmdlets" Version="7.1.1" />
    <PackageReference Include="Microsoft.PowerShell.Commands.Diagnostics" Version="7.1.1" />
    <PackageReference Include="Microsoft.PowerShell.SDK" Version="7.1.1" />
    <PackageReference Include="Microsoft.WSMan.Management" Version="7.1.1" />
    <PackageReference Include="System.Management.Automation" Version="7.1.1" />

Create an execution of PS7 code that generates an error.  e.g.:
        PowerShell ps = PowerShell.Create();
        ps.AddScript("Stop-Process -id 13"); 
        ps.Invoke(); 

Deploy app to IIS on Windows 2019 Core server

Enable Transcription logging on the IIS server.

## Expected behavior

Should see the error in transcription log and web app output from error stream:
```
PS>Stop-Process -id 13
**********************
PowerShell transcript start
Start time: 20210208153005

Configuration Name: 

Process ID: 8236
PSVersion: 7.1.1
PSEdition: Core
GitCommitId: 7.1.1
OS: Microsoft Windows 10.0.17763
Platform: Win32NT
PSCompatibleVersions: 1.0, 2.0, 3.0, 4.0, 5.0, 5.1.10032.0, 6.0.0, 6.1.0, 6.2.0, 7.0.0, 7.1.1
PSRemotingProtocolVersion: 2.3
SerializationVersion: 1.1.0.1
WSManStackVersion: 3.0
**********************
**********************
Command start time: 20210208153005
**********************
PS>CommandInvocation(Out-String): "Out-String"
>> ParameterBinding(Out-String): name="InputObject"; value="Cannot find a process with the process identifier 13."

Stop-Process: Cannot find a process with the process identifier 13.

Stop-Process: Cannot find a process with the process identifier 13.
```

## Actual behavior

Instead, no error output is shown in transcript or web app:
```
PS>Stop-Process -id 13
**********************
PowerShell transcript start
Start time: 20210208152932

Configuration Name: 

Process ID: 2732
PSVersion: 7.1.1
PSEdition: Core
GitCommitId: 7.1.1
OS: Microsoft Windows 10.0.17763
Platform: Win32NT
PSCompatibleVersions: 1.0, 2.0, 3.0, 4.0, 5.0, 5.1.10032.0, 6.0.0, 6.1.0, 6.2.0, 7.0.0, 7.1.1
PSRemotingProtocolVersion: 2.3
SerializationVersion: 1.1.0.1
WSManStackVersion: 3.0
**********************
**********************
PowerShell transcript start
Start time: 20210208152936

Configuration Name: 

Process ID: 2732
PSVersion: 7.1.1
PSEdition: Core
GitCommitId: 7.1.1
OS: Microsoft Windows 10.0.17763
Platform: Win32NT
PSCompatibleVersions: 1.0, 2.0, 3.0, 4.0, 5.0, 5.1.10032.0, 6.0.0, 6.1.0, 6.2.0, 7.0.0, 7.1.1
PSRemotingProtocolVersion: 2.3
SerializationVersion: 1.1.0.1
WSManStackVersion: 3.0
**********************
```

## Environment data

```

Name                           Value
----                           -----
PSVersion                      7.1.1
PSEdition                      Core
GitCommitId                    7.1.1
OS                             Microsoft Windows 10.0.17763
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

If I configure the PowerShell command to use "throw 'some error'" it works, but any other error method does not give error output. Transcription logging and standard output works fine otherwise, just errors are causing problems.  Once I turn off transcription it works as expected.

**Comment by pcarnella** [comment]
These are the PowerShell packages in the .Net app:

"Microsoft.Management.Infrastructure" Version="2.0.0" 
"Microsoft.Management.Infrastructure.CimCmdlets" Version="7.1.1" 
"Microsoft.PowerShell.Commands.Diagnostics" Version="7.1.1" 
"Microsoft.PowerShell.SDK" Version="7.1.1" 
"Microsoft.WSMan.Management" Version="7.1.1" 
"System.Management.Automation" Version="7.1.1"

**Comment by pcarnella** [comment]
Do you know where in the code base this would be? I can try to look myself.  It's causing me a lot of pain, preventing me from being able to effectively host PS 7 in my web app.

**Comment by iSazonov** [comment]
> Enable Transcription logging on the IIS server.

Do you mean PowerShell transcription logging?

**Comment by pcarnella** [comment]
Yes, PowerShell transcription logging.

**Comment by theJasonHelmick** [comment]
Hi @pcarnella -- The interactive WG is reviewing this issue and it will require further investigation. IS this still an ongoing issue? How did you enable the transcription logging for your ASP.net app on IIS?

**Comment by theJasonHelmick** [comment]
This issue has been closed waiting on author response - please reopen if this still needs to be addressed.

**Comment by pcarnella** [comment]
Hi, it was fixed in 7.3 thanks.  The workaround was to force all errors to be terminating, like wrap everything in try{}catch{throw} with erroraction stop.

________________________________
